test/capacitors/CapacitorFactory.t.sol

```diff
@@ -75,10 +76,17 @@ contract CapacitorFactoryTest is Setup {
 
         _cf.rescueFunds(address(_token), _raju, amount);
 
-        hoax(_owner);
+        hoax(_socketOwner);
         _cf.rescueFunds(address(_token), _raju, amount);
 
         uint256 finalBalOfOwner = _token.balanceOf(_raju);
         assertEq(finalBalOfOwner - initialBalOfOwner, amount);
     }
+
+    function testRescueNativeFunds() public {
+        uint256 amount = 1e18;
+
+        hoax(_socketOwner);
+        _rescueNative(address(_cf), NATIVE_TOKEN_ADDRESS, _fundRescuer, amount);
+    }
```

Comment:
Rescue cases need to be part of each contract. Can we use some inheritance pattern so that it happens automatically ?

Reply:
we will need some common contract object to be able to test like this. For now, the common code is in Setup file if that works.